### ISSUE TITLE
Could not get $batch with POST working

### ISSUE THREAD

**Issue by sudhakars4tf** [diegomvh/angular-odata#14]
When I use the following code I could not get below batch operation working (please note that there are **_NO issues/errors_** when I call `entities().get()` in my batch operation):

```
this.subscription.add(
	batch
	  .post(
		() => {
		  this.subscription.add(
			this.odataProfile
			 .entity()
                         .post(this.frm.value, {})
			  .subscribe((_resp: any) => {
				console.log(_resp);
			  }),
		  );  
		},
		{ responseType: 'entities' },
	)
	.subscribe((_resp: any) => {
		console.log(_resp);
		}),
	);
```

I am getting the following error:

`The message header '--changeset_kg5wzkmm--' is invalid. The header value must be of the format '<header name>: <header value>'.`

Any idea what might be causing this? 
Also, I could not figure out how to pass HttpOptions in the entity().put() method above....
Any help is gratefully appreciated!

Thank you!

**Comment by diegomvh** [comment]
Hi @sudhakars4tf 
Thanks for giving the library a try
When I tested the batch calls, I frankly did it working with GET requests, so this is probably an error, from what I see the request is wrong, could you paste the generated POST request? Thank you
It's strange, the options in the `post` entity request are optional, what version of the library are you using?

Regards

**Comment by diegomvh** [comment]
Fix #3ca6849
